homeassistant/components/eq3btsmart/climate.py

```diff
@@ -0,0 +1,390 @@
+        """Set new target hvac mode."""
+
+        if hvac_mode == HVACMode.OFF:
+            self._target_temperature = EQ3BT_OFF_TEMP
+            await self.async_set_temperature(temperature=EQ3BT_OFF_TEMP)
+
+        previous_mode = self.hvac_mode
+        self._attr_hvac_mode = hvac_mode
+
+        self.async_schedule_update_ha_state()
+
+        try:
+            await self._thermostat.async_set_mode(HA_TO_EQ_HVAC[hvac_mode])
+        except Eq3Exception:
+            _LOGGER.error("[%s] Failed setting HVAC mode", self._eq3_config.name)
+            self._attr_hvac_mode = previous_mode
+
+        self.async_schedule_update_ha_state()
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode."""
+
+        if self._thermostat.status is None:
+            return None
+
+        if self._thermostat.status.is_window_open:
+            return Preset.WINDOW_OPEN
+        if self._thermostat.status.is_boost:
+            return Preset.BOOST
+        if self._thermostat.status.is_low_battery:
+            return Preset.LOW_BATTERY
+        if self._thermostat.status.is_away:
+            return Preset.AWAY
+        if self._thermostat.status.operation_mode == OperationMode.ON:
+            return Preset.OPEN
+
+        if self._thermostat.status.presets is None:
+            return None
+
+        if (
+            self._thermostat.status.target_temperature
+            == self._thermostat.status.presets.eco_temperature
+        ):
+            return Preset.ECO
+        if (
+            self._thermostat.status.target_temperature
+            == self._thermostat.status.presets.comfort_temperature
+        ):
+            return Preset.COMFORT
+        return PRESET_NONE
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+
+        match preset_mode:
+            case Preset.BOOST:
+                await self._thermostat.async_set_boost(True)
+            case Preset.AWAY:
+                await self._thermostat.async_set_away(True)
+            case Preset.ECO:
+                await self._thermostat.async_set_preset(Eq3Preset.ECO)
+            case Preset.COMFORT:
+                await self._thermostat.async_set_preset(Eq3Preset.COMFORT)
+            case Preset.OPEN:
+                await self._thermostat.async_set_mode(OperationMode.ON)
+
+        if self._thermostat.status is not None:
+            self._target_temperature = self._thermostat.status.target_temperature.value
+
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return device information."""
+
+        return DeviceInfo(
+            name=self._eq3_config.name,
+            manufacturer=MANUFACTURER,
+            model=DEVICE_MODEL,
+            identifiers={(DOMAIN, self._eq3_config.mac_address)},
+            sw_version=self._firmware_version,
+            connections={(CONNECTION_BLUETOOTH, self._eq3_config.mac_address)},
+        )
```

Comment:
personal preference: set it thorugh `_attr_device_info` as it's static

Reply:
True, I only found out about that attribute today, will use that.